messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java

```diff
@@ -161,8 +162,41 @@ public ActiveMQServerService(Configuration configuration, PathConfig pathConfig)
         return dataSource;
     }
 
-    Map<String, JChannel> getChannels() {
-        return channels;
+    JChannel getChannel(String name) throws Exception {
+        if (channels.containsKey(name)) {
+            return channels.get(name);
+        }
+        if (jgroupsChannels.containsValue(name)) {
+            Optional<String> groupName = jgroupsChannels.entrySet().stream().filter(entry -> name.equals(entry.getValue())).map(entry -> entry.getKey()).findFirst();
+            if (groupName.isPresent()) {
+                final String group;
+                if(groupName.get().startsWith("discovery") || groupName.get().startsWith("broadcast")) {
+                    group = groupName.get().substring(9);
+                } else {
+                    group = groupName.get();
+                }
+                if (configuration.getDiscoveryGroupConfigurations().containsKey(group)) {
+                    try {
+                        configuration.getDiscoveryGroupConfigurations().get(group).getBroadcastEndpointFactory().createBroadcastEndpoint();
```

Comment:
Unless I'm mistaken, this is going to mess with the JChannelWrapper reference counting mechanism, and will result in the channel not closing when it should.